_posts/2019-12-24-templatowanie-jobow-jenkinsa.md

```diff
@@ -0,0 +1,114 @@
+---
+layout:    post
+title:     "Automat dodający joby do Jenkinsa"
+published: false
+date:      2019-12-24 08:00:00 +0100
+author:    dkubiak
+tags:
+    - jenkins job
+    - jenkins template
+    - devops
+    - jenkins-job-builder
+    - jenkins-jobs
+---
+## Zastosowanie
+
+W ogarniającym nas świecie mikroserwisów skala projektów do utrzymania staje sie ogromna. Każdy z tych projektów musimy przecież: zbudować, przetesować, zdeployować itd. Przy liczbie projektów 20+ przestaje to być trywialne. W tym artukule zajmniemy się pierwszym zagadnieniem budowaniem, ale opisany tutaj sposób bez problemu można zastosować do innych aspektów.
+
+Do budowania naszych projektów 20+ zazwyczaj staramy sie utrzymać jedno narzędzie i to oczywiście działa na plus. W tym artykule posługiwać będziemy się mavenem.
+
+Do zautomatyzowania naszego procesu posłuży nam jenkins.
+
+Przejdzy do sedna, czyli jak budować projekty 20+ z jak najmniejszym nakładem parcy i ilością kodu do utrzymania.
+
+## Startujemy
+
+Przy założeniu, że nasze projekty budujemy
+
+`mvn clean package`
+
+-/+ jakieś super ważne przełączniki typu` -DskipTes...` ;) jesteśy w stanie w bardzo prosty i schludy sposób, zbudować kod/konfigurację która zautomatyzuje nasz cały proces.
+
+Automatyzajcę rozpoczniemy od użycia narzędzia: [jenkins-job-builder](https://docs.openstack.org/infra/jenkins-job-builder/ "jenkins-job-builder")
+
+Instalacja: `pip install --user jenkins-job-builder`
+macOS `brew install jenkins-job-builder`
+
+Definiujemy plik konfiguracyjny dla jenkins-jobs w lokalizacji `/etc/jenkins_jobs/jenkins_jobs.ini`:
+
+    [jenkins]
+    query_plugins_info=False
+    user=jenkins #Użytkownik Jenkinsowy
+    password=93a1160c11dc014b7214d4e8769fe8c9
+    url=http://localhost:8080 #Url do jenkinsa
+
+- user - Użytkownik Jenkinsowy
+- password - API Token dla swojego użytkownika [link](https://support.cloudbees.com/hc/en-us/articles/115003090592-How-to-re-generate-my-Jenkins-user-token)
+- url - Adres URL do Jenkinsa
+
+Tak skonfigurowane narzędzie pozwoli nam utworzy dowolny job jenkinsowy.
+Utworz plik o nazwie `project1-build.yaml` w katalogu `jobs` z zawartością
+
+    - job:
+        name: project-1-build
+        project-type: freestyle
+        disabled: false
+        builders:
+            - shell: 'mvn clean package'
+
+Zasilenie jenkinsa nowo utworzonym jobem:
+`jenkins-jobs update jobs`
+
+W ten sposob dodałeś pierwszy z 20+ projektów do jenkinsa. Good Job!
+
+## Szablony
+
+Oczywiście! szablony, uwielbiamy opakowywać wszystko w pewne wzorce, wspólne procesy i nie może zabraknąć tutaj słynnego "re-użycia". 
```

Comment:
"Oczywiście! szablony, uwielbiamy" Wielka litera po kropce, ale jeszcze lepiej byłoby to przeredagować. 
reużycie, bez myślnika

Reply:
done!